commitizen/commands/bump.py

```diff
@@ -207,17 +206,24 @@ def __call__(self) -> None:  # noqa: C901
                     "--local-version cannot be combined with --build-metadata"
                 )
 
-        # If user specified changelog_to_stdout, they probably want the
-        # changelog to be generated as well, this is the most intuitive solution
-        self.changelog = self.changelog or bool(self.changelog_to_stdout)
-
         if get_next:
-            if self.changelog:
+            # if trying to use --get-next, we should not allow --changelog or --changelog-to-stdout
+            if self.changelog_flag or bool(self.changelog_to_stdout):
                 raise NotAllowed(
                     "--changelog or --changelog-to-stdout is not allowed with --get-next"
                 )
+            # --get-next is a special case, taking precedence over config for 'update_changelog_on_bump'
+            self.changelog_config = False
             # Setting dry_run to prevent any unwanted changes to the repo or files
             self.dry_run = True
+        else:
+            # If user specified changelog_to_stdout, they probably want the
+            # changelog to be generated as well, this is the most intuitive solution
```

Comment:
Hmmm... I think otherwise. If the user wants it to be printed, they might want to take a look first before making the change

Reply:
This comment and the associated self.changelog_flag = (self.changelog_flag or bool(self.changelog_to_stdout) or self.changelog_config) is from the L210-L212 of the previous code. (with the inclusion of the new `changelog_config` change)

Reply:
>  If a user wants to print something, they may want to review it first before making any changes.

And, from what I understand from the the rest of the `bump` code, using `changelog_to_stdout` flag essentially put the changelog generation in dry run. So it will not change anyfile (my MR doesn't seem to change that behavior I guess)